### PR TITLE
HAI-1987 Separate authorization and business logic in application attachments

### PR DESCRIPTION
**Pull request by corvidian**
# Description

Authorization and business logic have been intertwined in application attachments. Authorization has checked that the user has the correct permission for the application ID in the URL, but checking that the attachment belongs to that application has been done in ApplicationAttachmentService.

Separate authorization and business logic by adding attachment authorization to ApplicationAuthorizer. The authorizer checks that the user has the correct permission for the application's hanke and that the attachment belongs to that application. We don't need to pass the applicationId to the service anymore.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [X] Other

# Instructions for testing
Application attachment authorization should no have changed.

# Checklist:

- [X] I have written new tests (if applicable)
- [X] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
 or other location: 

# Other relevant info
The actual functionality of HAI-1987 is coming in another PR.

### CODE REVIEW

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationAuthorizer.kt [hunk 1]
```diff
@@ -29,4 +36,18 @@ class ApplicationAuthorizer(
...
+
+    @Transactional(readOnly = true)
+    fun authorizeAttachment(
+        applicationId: Long,
+        attachmentId: UUID,
+        permissionCode: String
+    ): Boolean {
+        authorizeApplicationId(applicationId, permissionCode)
+        if (findApplicationIdForAttachment(attachmentId) == applicationId) return true
+        throw AttachmentNotFoundException(attachmentId)
```
Comment (umeetiusbaar) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationAuthorizer.kt:48]: Would it be more clear if `return if` is used? 
```
return if (findApplicationIdForAttachment(attachmentId) == applicationId) true else throw AttachmentNotFoundException(attachmentId)
```

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentServiceITest.kt [hunk 2]
```diff
@@ -379,10 +345,7 @@ class ApplicationAttachmentServiceITest : DatabaseTest() {
             )
         assertThat(applicationAttachmentRepository.findById(attachment.id)).isPresent()
 
-        applicationAttachmentService.deleteAttachment(
-            applicationId = application.id!!,
-            attachmentId = attachment.id
-        )
+        applicationAttachmentService.deleteAttachment(attachmentId = attachment.id)
```
Comment (umeetiusbaar) [inline services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentServiceITest.kt:348]: Is parameter naming necessary?

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentServiceITest.kt [hunk 3]
```diff
@@ -128,46 +127,13 @@ class ApplicationAttachmentServiceITest : DatabaseTest() {
                 attachment = file
             )
 
-        val result =
-            applicationAttachmentService.getContent(
-                applicationId = application.id!!,
-                attachmentId = attachment.id
-            )
+        val result = applicationAttachmentService.getContent(attachmentId = attachment.id)
```
Comment (umeetiusbaar) [inline services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentServiceITest.kt:130]: Is parameter naming necessary?
```
applicationAttachmentService.deleteAttachment(attachment.id)
```

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentServiceITest.kt (hunk 2 above)
Comment (corvidian) [reply services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentServiceITest.kt]: Thank you, removed the parameter name.

**Pull request merged**